block-producer/src/batch_builder/mod.rs

```diff
@@ -18,66 +25,105 @@ mod tests;
 ///
 /// Note: Until recursive proofs are available in the Miden VM, we don't include the common proof.
 pub struct TransactionBatch {
-    txs: Vec<SharedProvenTx>,
+    account_initial_states: Vec<(AccountId, Digest)>,
+    updated_accounts: Vec<(AccountId, Digest)>,
+    consumed_notes_script_roots: Vec<Digest>,
```

Comment:
Do we still need the note script roots? I thought we the latest refactoring in miden base we no longer need this.

Reply:
removed